azureclient/basicfactory/spec.go

```diff
@@ -0,0 +1,31 @@
+package basicfactory
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/v2/pkg/apis/provider/v1alpha1"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-11-01/network"
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-05-01/resources"
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2019-04-01/storage"
+
+	"github.com/giantswarm/azure-operator/v5/azureclient/credentialprovider"
+)
+
+// The basicfactory.Interface interface defines an interface to return an Azure API Client given the
+// authentication data provided via a AzureClientCredentialsConfig object.
+// Implementations of this interface are not meant to be run standalone, but to be used by a credentialawarefactory.Interface.
+type Interface interface {
```

Comment:
What if the name of this package was `basic` and this interface was `Factory` - then it would be `basic.Factory` under `azureclient`. WDYT?

Reply:
what about the default implementation? Now it's called BasicFactory how do you suggest to name it?

Reply:
Up to you. When there's an interface that is expected to be used, I'd just create some `New()` function which constructs it and the implementation can be anything. `basicFactory` or `factory` maybe?

Reply:
ok makes sense. Ensured I use the interface as the return method and renamed to `factory`